ios/FluentUI/BottomSheet/BottomSheetViewController.swift

```diff
@@ -0,0 +1,404 @@
+            let targetFrame: CGRect
+            let futureDimmingViewAlpha: CGFloat
+            switch currentState {
+            case .collapse:
+                targetFrame = targetExpandFrame(with: window)
+                setupDimmingView()
+                futureDimmingViewAlpha = 1
+            case .expand:
+                targetFrame = targetCollapseFrame(with: window)
+                futureDimmingViewAlpha = 0
+            }
+
+            animator = UIViewPropertyAnimator(duration: Constants.animationDuration, curve: .linear, animations: { [weak self] in
+                self?.view.frame = targetFrame
+                self?.dimmingView.alpha = futureDimmingViewAlpha
+            })
+        }
+    }
+
+    private func addAnimatorCompletion(to futureState: BottomSheetViewState, completion: (() -> Void)? = nil) {
+        animator?.addCompletion { [weak self] _ in
+            if let currentBottomSheetViewController = self {
+                currentBottomSheetViewController.panGestureRecognizer.isEnabled = true
+                currentBottomSheetViewController.currentState = futureState
+                currentBottomSheetViewController.view.layoutIfNeeded()
+
+                // when the bottomsheet drawer is expanded, we don't want the UIViews behind the sheet to be accessible.
+                currentBottomSheetViewController.view.accessibilityViewIsModal = futureState == .expand
+                currentBottomSheetViewController.updateResizingHandleViewAccessibility()
+                UIAccessibility.post(notification: .layoutChanged, argument: currentBottomSheetViewController.resizingHandleView)
+
+                completion?()
+                if futureState == .expand {
+                    currentBottomSheetViewController.delegate?.bottomSheetViewControllerDidExpand?(currentBottomSheetViewController)
+                } else {
+                    currentBottomSheetViewController.removeDimmingView()
+                    currentBottomSheetViewController.delegate?.bottomSheetViewControllerDidCollapse?(currentBottomSheetViewController)
+                }
+            }
+        }
+    }
+
+// MARK: - pan gesture utilities
+
+    @objc private func handlePan(gesture: UIPanGestureRecognizer) {
+        let positionInSuperview = gesture.translation(in: view.superview)
+
+        switch gesture.state {
+        case .began:
+            panningStart()
+        case .changed:
+            panningProgress(in: positionInSuperview.y)
+        case .ended:
+            let velocity = gesture.velocity(in: view)
+            panningStop(in: positionInSuperview.y, velocity: velocity.y)
+        default:
+           break
+        }
+    }
+
+    private func panningStart() {
+        if animator?.isRunning ?? false {
+           return
+        }
+
+        setupAnimator()
+    }
+
+    private func panningStop(in verticalPos: CGFloat, velocity: CGFloat) {
+        panGestureRecognizer.isEnabled = false
```

Comment:
When I swipe ~10 times, I can get the sheet in a state where it's collapsed and the animation is over, but the panGestureRecognizer is permanently stuck in a disabled state so I can't swipe anymore. Only way to get it unstuck is to click the resize handle. Maybe the animation completion handler isn't reliable?